### PR TITLE
Stubs for Classes

### PR DESCRIPTION
**Pull request by haven1433**
Main changes:
* Support for making Stubs from classes, not just interfaces.
* Class Stubs provide access to all protected members.
* Class Stubs provide access to any implementations from the base class (important for virtual members).
* Class Stubs allow you to assign implementations before calling the baseclass constructor (important for classes that call abstract or virtual members during their constructor).

Additional changes:
* Added "usage tests" as examples, showing how to use AutoImplement's Stubs in various ways.
* Added framework tests showing how .net behaves in some edge cases, so we can design around those.
* Composite members are now virtual, which makes it easier to make custom composite objects by extending from AutoImplement's result.
* bug fix: stubs for interfaces that have base-interfaces with events now compile.
* bug fix: generic methods with custom implementation work correctly now.

### CODE REVIEW

File: src/AutoImplement.Tests.Types/PoorlyImplementedAbstractClass.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Linq;
+
+namespace HavenSoft.AutoImplement.Tests.Types {
+   // It's possbile for an abstract class to call an abstract method during the constructor.
```
Comment (rtzoeller) [inline src/AutoImplement.Tests.Types/PoorlyImplementedAbstractClass.cs]: Spelling: possible

**Review by rtzoeller — APPROVED** [review]
Looks good!
